### PR TITLE
Improvement/ HOLO 595 Enforce prettier / lint run on protocol

### PR DESCRIPTION
**Pull request by BravoNatalie**
## Describe Changes

- added `workflows/enforce_prettier.yml`
- added `workflowsqenforce_eslint.yml`
- added eslint configuration

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

File: .github/workflows/enforce_prettier.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          fetch-depth: 0
+          ref: ${{ github.head_ref }}
+          # Make sure the value of GITHUB_TOKEN will not be persisted in repo's config
+          persist-credentials: false
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v4.2
+        with:
+          prettier_options: --check deploy/**/*.ts test/**/*.ts scripts/**/*.ts
+          dry: True
```
Comment (alexanderattar) [inline .github/workflows/enforce_prettier.yml:34]: does this mean?

**Review by alexanderattar — APPROVED** [review]
Cool this will be nice to have

**Review by alexanderattar — CHANGES_REQUESTED** [review]
image

There's a lot of options in the prettier config that are being ignored. I think we can remove those. Also this is currently running the prettier action in as a dry-run. I believe we need to set `dry` to false so it actually runs the changes

**Comment by alexanderattar** [comment]
On further inspection of the previously run action, I noticed some issues. See above ☝️

File: .github/workflows/enforce_prettier.yml (hunk 1 above)
Comment (BravoNatalie) [reply .github/workflows/enforce_prettier.yml:34]: it runs the prettier check on the .ts files under this folders

**Comment by BravoNatalie** [comment]
> image
> 
> There's a lot of options in the prettier config that are being ignored. I think we can remove those. Also this is currently running the prettier action in as a dry-run. I believe we need to set `dry` to false so it actually runs the changes

This's happening because the prettier file has these rules, I haven't removed them because I thought the file was created like that on purpose. But if you agree I can remove them.

**Comment by alexanderattar** [comment]
Yep those rules are my bad. I removed them and pushed the change

**Review by alexanderattar — APPROVED** [review]

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**